ray-operator/test/e2e/rayservice_in_place_update_test.go

```diff
@@ -0,0 +1,110 @@
+package e2e
+
+import (
+	"path"
+	"strings"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+	"github.com/ray-project/kuberay/ray-operator/test/sampleyaml"
+	. "github.com/ray-project/kuberay/ray-operator/test/support"
+)
+
+func TestRayServiceInPlaceUpdate(t *testing.T) {
+	test := With(t)
+	g := NewWithT(t)
+
+	// Create a namespace
+	namespace := test.NewTestNamespace()
+	test.StreamKubeRayOperatorLogs()
+
+	fileName := "ray-service.sample.yaml"
+
+	yamlFilePath := path.Join(sampleyaml.GetSampleYAMLDir(test), fileName)
+	rayServiceFromYaml := DeserializeRayServiceYAML(test, yamlFilePath)
+	KubectlApplyYAML(test, yamlFilePath, namespace.Name)
+
+	rayService, err := GetRayService(test, namespace.Name, rayServiceFromYaml.Name)
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(rayService).NotTo(BeNil())
+
+	var rayClusterName string
+	// Wait for RayCluster name to be populated
+	g.Eventually(func(g Gomega) {
+		rs, err := GetRayService(test, namespace.Name, rayServiceFromYaml.Name)
+		g.Expect(err).NotTo(HaveOccurred())
+		if rs.Status.PendingServiceStatus.RayClusterName != "" {
+			rayClusterName = rs.Status.PendingServiceStatus.RayClusterName
+		} else if rs.Status.ActiveServiceStatus.RayClusterName != "" {
+			rayClusterName = rs.Status.ActiveServiceStatus.RayClusterName
+		}
+		g.Expect(rayClusterName).NotTo(BeEmpty())
+	}, TestTimeoutShort).Should(Succeed())
+
+	test.T().Logf("Waiting for RayCluster %s/%s to be ready", namespace.Name, rayClusterName)
+	g.Eventually(RayCluster(test, namespace.Name, rayClusterName), TestTimeoutMedium).
+		Should(WithTransform(RayClusterState, Equal(rayv1.Ready)))
+	rayCluster, err := GetRayCluster(test, namespace.Name, rayClusterName)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	// Create curl pod
+	curlPodName := "curl-pod"
+	curlContainerName := "curl-container"
+
+	curlPod, err := CreateCurlPod(test, curlPodName, curlContainerName, namespace.Name)
```

Comment:
We should wait until the curl pod becomes ready using `Eventually`

Reply:
I check the curl pod status below the create curl pod

Reply:
Oh I found it. I think you can put it directly under this line to improve readability.